app/lib/common/dialogs/deactivation_confirmation.dart

```diff
@@ -0,0 +1,124 @@
+import 'package:acter/common/utils/routes.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_riverpod/flutter_riverpod.dart';
+import 'package:go_router/go_router.dart';
+
+// Can be extended to be reusable dialog as riverpod states get added.
+// Ref can be used to read any provider which are declared.
+void deactivationConfirmationDialog(BuildContext context, WidgetRef ref) {
+  TextEditingController passwordController = TextEditingController();
+  showDialog(
+    context: context,
+    builder: (BuildContext ctx) {
+      return AlertDialog(
+        title: Text('Deactivate Account',
+            style: TextStyle(
+                color: AppTheme.brandColorScheme.error, fontSize: 32,),),
+        content: SingleChildScrollView(
+          child: ConstrainedBox(
+            constraints: const BoxConstraints(maxWidth: 650),
+            child: Wrap(
+              children: [
+                Text(
+                  'Careful: You are about to permanently deactivate your account ',
+                  style: TextStyle(
+                    color: AppTheme.brandColorScheme.error,
+                    fontSize: 24,
+                  ),
+                ),
+                const Text('If you proceed: \n  \n'
+                    '- All your personal data will be removed from your homeserver, including display name and avatar \n'
+                    '- All your sessions will be closed immediately, no other device will be able to continue their sessions \n'
+                    '- You will leave all rooms, chats, spaces and DMs that you are in \n'
+                    '- You will not be able to reactivate your account \n'
+                    '- You will no longer be able to log in \n'
+                    '- No one will be able to reuse your username (MXID), including you: this username will remain unavailable indefinitely \n'
+                    '- You will be removed from the identity server, if you provided any information to be found through that (e.g. email or phone number) \n'
+                    '- All local data, including any encryption keys, will be permanently deleted from this device \n'
+                    '- Your old messages will still be visible to people who received them, just like emails you sent in the past. \n'
+                    '\n You will not be able to reverse any of this. This is a permanent and irrevocable action.'),
+                Padding(
+                  padding: const EdgeInsets.only(top: 10, bottom: 10),
+                  child: Text(
+                    'Please provide your user password to confirm you want to deactivate your account.',
+                    style: TextStyle(
+                      color: AppTheme.brandColorScheme.error,
+                      fontSize: 24,
+                    ),
+                  ),
+                ),
+                ConstrainedBox(
+                  constraints: const BoxConstraints(maxWidth: 250),
```

Comment:
I feel like ```MediaQuery``` width should be used here. The content should expand itself based on current context.

Reply:
the reason I bounded it is that it would normally expand to full width (+ some padding on the sides) but that makes the text pretty unreadable (12-words rule). Using a max-width I can assure we never overexpand, but do contract accordingly on smaller devices. What would be the benefit to look at mediaquery and do it in relation to that?

Reply:
Thanks for reasoning, I understand it now.